### DISCUSSION TITLE
How to activate and use H264 , H265 in RustDesk nightly release

### DISCUSSION THREAD

**Discussion by mikekgr** [rustdesk/rustdesk#4095] [Q&A] [answered]
Dear Sirs, Dear @rustdesk 
I am using RustDesk to connect Windows 10 computers having installed the nightly release. Loocking at the RustDesk settings => General => Display Default Codec , in both computers there are listed both H264 & H265. After the connection activating the Show quality monitor , the activated codec is VP9 and in Codec submenu the possible choices are : Auto , VP9 , VP8 and deactive the H264 , H265 ... I don't know how to activate either H264 or H265 please help me.

Thanks and Best Regards,
Mike Kranidis

**Comment by rustdesk** [comment]
@21pages why is it shown in settings on both sides but inactive on remote menubar?

But current hardware codec implementation is bad, can not give you better experience except lower cpu usage. We are still polishing it.

**Comment by 21pages** [comment, marked as answer]
> Display Default Codec 
in both computers there are listed both H264 & H265

both support H264/H265 decoding
> Codec submenu
deactive the H264 , H265

The controlled end doesn't support H264/H265 encoding.

**Reply by mikekgr** [reply]
Dear @21pages ,
thanks a lot for your answer. I have a possible misunderstanding, when you said "The controlled end doesn't support H264/H265 encoding" what do you means? You think that my remote computer is not capable to support H264/H265 encoding? Is there a way to test it?
Thanks and Best Regards,
Mike Kranidis

**Reply by 21pages** [reply]
https://github.com/21pages/test/releases/download/test/hwcodec_available.exe

**Reply by mikekgr** [reply]
> https://github.com/21pages/test/releases/download/test/hwcodec_available.exe

Thank you very much for the test program. I just run it to my laptop (I will run it in more PC later on) and I got:
U:\Downloads>hwcodec_available.exe
[2023-04-19T03:02:32Z INFO  available] encoder count:0
[2023-04-19T03:02:32Z INFO  available] decoders count:6
"h264" AV_HWDEVICE_TYPE_NONE
"hevc" AV_HWDEVICE_TYPE_NONE
"h264" AV_HWDEVICE_TYPE_DXVA2
"h264" AV_HWDEVICE_TYPE_D3D11VA
"h264_qsv" AV_HWDEVICE_TYPE_NONE
"h264_qsv" AV_HWDEVICE_TYPE_QSV

that is looking not available H264 , H265 hardware ecoders...

Best Regards,
Mike Kranidis
